pkg/cloudprovider/cloudprovider.go

```diff
@@ -87,6 +88,10 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("resolving node class, %w", err))
 	}
 
+	// TODO: Remove this once support for conversion webhooks is dropped
+	if nodeClass.UbuntuIncompatible() {
+		return nil, fmt.Errorf("EC2NodeClass %q is incompatible with Karpenter v1, specify your Ubuntu AMIs in your AMISelectorTerms", nodeClass.Name)
```

Comment:
Should this be a NodeClassNotReady error that should cause us to delete the NodeClaim? The only time this error can occur is when we have created a NodeClaim prior to the upgrade and then we resolve it after fact -- otherwise the NodePool is going to go NotReady after some time. I think we are good to just delete immediately since newly created NodeClaims shouldn't be using NodePools that have NodeClasses that aren't ready